### ISSUE TITLE
Error in PredictCost() for the op: "Softmax"

### ISSUE THREAD

**Issue by ddaspit** [OpenNMT/OpenNMT-tf#838]
- Version: 2.19.0
- Tensorflow: 2.5.0
- CUDA: 11.2.0
- cuDNN: 8.1.0
- Platform: Windows 10
- GPU: Nvidia RTX 3090

I am trying to train a Transformer model using OpenNMT-tf using the following command:
```
onmt-main --model_type Transformer --config config.yml --auto_config train --with_eval
```
When it starts to evaluate the model, I receive the following error multiple times:
```
2021-06-09 16:29:34.868679: W tensorflow/core/grappler/costs/op_level_cost_estimator.cc:689] Error in PredictCost() for the op: op: "Softmax" attr { key: "T" value { type: DT_FLOAT } } inputs { dtype: DT_FLOAT shape { unknown_rank: true } } device { type: "GPU" vendor: "NVIDIA" model: "GeForce RTX 3090" frequency: 1695 num_cores: 82 environment { key: "architecture" value: "8.6" } environment { key: "cuda" value: "11020" } environment { key: "cudnn" value: "8100" } num_registers: 65536 l1_cache_size: 24576 l2_cache_size: 6291456 shared_memory_size_per_multiprocessor: 102400 memory_size: 22723493888 bandwidth: 936096000 } outputs { dtype: DT_FLOAT shape { unknown_rank: true } }
```
I am using a minimal configuration file that just configures the training/evaluation corpus and vocabulary. I was able to train without any issues using OpenNMT-tf 2.18.1 and Tensorflow 2.4.1.

**Comment by guillaumekln** [comment]
The log was added in TensorFlow 2.5. It is marked as a warning (see the "W") so I'm not sure this is actually an issue. Do you see any other issues like degraded performance or score when running the evaluation?

**Comment by ddaspit** [comment]
I think I was seeing some degraded performance, but I will need to check more thoroughly to be sure. I will post what I find here.

**Comment by guillaumekln** [comment]
Did you gather more information?

The log is not very helpful as it does not clearly say what the error actually is. The possible issue is that it reports `unknown_rank: true`, but I checked the graph that we build and the rank is actually known for this Softmax op.

**Comment by guillaumekln** [comment]
I identified the origin of the log and opened a TensorFlow issue: https://github.com/tensorflow/tensorflow/issues/50575

**Comment by ddaspit** [comment]
I haven't had a chance to move my machine back to Tensorflow 2.5.0 to test if there is any degraded performance.

**Comment by guillaumekln** [comment]
I'm closing this issue since I believe it should be fixed in TensorFlow. You can watch the TensorFlow issue I linked above.

**Comment by dyjzkl** [comment]
Do you know how this problem is caused? I also encountered this problem
